src/uint256.h

```diff
@@ -53,15 +53,27 @@ class base_blob
         std::fill(m_data.begin(), m_data.end(), 0);
     }
 
+    /** Lexicographic ordering
+     * @note Does NOT match the ordering on the corresponding \ref
+     *       base_uint::CompareTo, which starts comparing from the end.
+     */
     constexpr int Compare(const base_blob& other) const { return std::memcmp(m_data.data(), other.m_data.data(), WIDTH); }
 
     friend constexpr bool operator==(const base_blob& a, const base_blob& b) { return a.Compare(b) == 0; }
     friend constexpr bool operator!=(const base_blob& a, const base_blob& b) { return a.Compare(b) != 0; }
     friend constexpr bool operator<(const base_blob& a, const base_blob& b) { return a.Compare(b) < 0; }
 
-    // Hex string representations are little-endian.
+    /* Hex string representations in reverse byte order of the internal array.
+     * This is consistent with the internal array of arith_* being little-endian.
```

Comment:
In commit "doc + test: Correct uint256 hex string endianness" (07de45a882d43fd5567b2a6b7c8a2da3d6dbda3d)

Statement "This is consistent with the internal array of arith_* being little-endian." seems a bit vague to me. Also calling internal representation of the arithmetic class little endian might not be ideal because endianness normally refers to bytes and the arithmetic class stores an array of uint32_t numbers, not of bytes.

I think I would suggest changing line to:

- The reverse hex string representation can be a convenient way of showing the blob as a number, because it is consistent with the way the `base_uint` class converts blobs to numbers.

  `base_uint` treats the blob as an array of bytes with the least significant byte first, and most significant byte last. Because numbers are typically written with the most significant digit first, and least significant digit last, the reverse hex display of the blob corresponds to the same arithmetic number `base_uint` interprets from the blob.

Reply:
👍, I like that we're not saying endianness for arrays anymore (as suggested by @sipa as well), but seems we still do here. Would be nice if we could eventually unify throughout the codebase.